libraries/PWMAudio/src/PWMAudio.cpp

```diff
@@ -219,3 +245,33 @@ size_t PWMAudio::write(const uint8_t *buffer, size_t size) {
     }
     return writtenSize;
 }
+
+void PWMAudio::find_pacer_fraction(int target, uint16_t *numerator, uint16_t *denominator) {
+    const uint16_t max = 0xFFFF;
+
```

Comment:
This does look like it could take a little bit to find the right numerator/denominator.  It's the same problem as the SPI clock change (and others), and I think your method is the cleanest way.

That said. can we cache the last result for numerator/denomiator?  Maybe make something like
````
void PWMAudio::find_pacer_fraction(int target, uint16_t *numerator, uint16_t *denominator) {
    const uint16_t max = 0xFFFF;
    static int  lastTarget;
    static  uint16_t lastNum, lastDen;

    if (target==lastTarget) {
        *numerator = lastNum;
        *denominator = lastDen;
        return;
    }
....
<before exiting the function...>
    lastTarget = target;
    lastNum = bestNum;
    lastDen = bestDenom;

    *numerator = bestNum;
    *denominator = bestDenom;
}
````

I'm thinking that most use cases will have the same frequency if they do multiple PWMAudio samples.  So no need to recalc.  And, worst case, we use 64 bits of RAM and a negligible amount of CPU cycles

Reply:
I put 4 samples together in a video(Because i cant upload audio here). First two are after fix and last two before fix.
https://github.com/earlephilhower/arduino-pico/assets/9679272/4b2914d6-5eee-4e04-abee-97fcbb8285c5
Here, two spectrum graphs if it weren't obvious already (Or some got bad hearing)
1 is 8000hz samplerate before fix and 2 is 8000hz after.
![image](https://github.com/earlephilhower/arduino-pico/assets/9679272/ae19cfe2-50e6-4b2a-b193-fee8201e8d7c)
![image](https://github.com/earlephilhower/arduino-pico/assets/9679272/609b2bd0-a368-4326-ae5e-d5e9ded755ce)

Also thank you for the feedback! I will fix those right away and see what i can do.